src/types/types.ts

```diff
@@ -793,6 +824,217 @@ export interface ServiceAccountInput {
 }
 
+export interface Model3DListRequest {
+  limit?: number;
+  /**
+   * Filter based on whether or not it has published revisions.
+   */
+  published?: boolean;
+}
+
+export interface CreateRevision3D {
+  /**
+   * True if the revision is marked as published.
+   */
+  published?: boolean;
+  /**
+   * Global rotation to be applied to the entire model.
+   * The rotation is expressed by Euler angles in radians and in XYZ order.
+   */
+  rotation?: [boolean, boolean, boolean];
+  camera?: RevisionCameraProperties;
+  /**
+   * The file id to a file uploaded to Cognite's Files API.
+   * Can only be set on revision creation, and can never be updated. _Only FBX files are supported_.
+   */
+  fileId: CogniteInternalId;
+}
+
+export type Tuple3<T> = [T, T, T];
+
+export interface UpdateRevision3D {
+  id: CogniteInternalId;
+  update?: {
+    /**
+     * True if the revision is marked as published.
+     */
+    published?: SetField<boolean>;
+    /**
+     * Global rotation to be applied to the entire model.
+     * The rotation is expressed by Euler angles in radians and in XYZ order.
+     */
+    rotation?: SetField<Tuple3<number>>;
+    /**
+     * Initial camera target.
+     */
+    camera?: SetField<RevisionCameraProperties>;
+  };
+}
+
+export interface Revision3DListRequest {
+  /**
+   * Limits the number of results to be returned.
+   * The maximum results returned by the server is 1000 even if the limit specified is larger.
+   */
+  limit?: number;
+  /**
+   * Filter based on whether or not it has published revisions.
+   */
+  published?: boolean;
+}
+
+export interface RevisionCameraProperties {
+  /**
+   * Initial camera target.
+   */
+  target?: [number, number, number];
+  /**
+   * Initial camera position.
+   */
+  position?: [number, number, number];
+}
+
+export interface Revision3D {
+  /**
+   * The ID of the revision.
+   */
+  id: CogniteInternalId;
+  /**
+   * The file id.
+   */
+  fileId: CogniteInternalId;
+  /**
+   * True if the revision is marked as published.
+   */
+  published: boolean;
+  /**
+   * Global rotation to be applied to the entire model.
+   * The rotation is expressed by Euler angles in radians and in XYZ order.
+   */
+  rotation?: [number, number, number];
+  camera?: RevisionCameraProperties;
+  /**
+   * The status of the revision.
+   */
+  // tslint:disable-next-line
+  status: 'Queued' | 'Processing' | 'Done' | 'Failed';
+  /**
+   * The threed file ID of a thumbnail for the revision.
+   * Use /3d/files/{id} to retrieve the file.
+   */
+  thumbnailThreedFileId?: number;
+  /**
+   * The URL of a thumbnail for the revision.
+   */
+  thumbnailURL?: string;
+  /**
+   * The number of asset mappings for this revision.
+   */
+  assetMappingCount: number;
+  /**
+   *
+   */
+  createdTime: Date;
+}
+
+export interface Limit {
+  /**
+   * Limits the number of results to be returned.
+   * The maximum results returned by the server is 1000 even if the limit specified is larger.
```

Comment:
I don't think this is true anymore. If more then server will return `400`

Reply:
it's in spec
should I remove it?

Reply:
Yes, please